From c8f827309ea7223dacbb2f155ba1239eb3195312 Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Fri, 25 Jun 2021 11:31:23 +0200
Subject: [PATCH] Add video driver for sq930x

This should be send to upstream as it is verified that it works.
It was requested here:
https://gitlab.nic.cz/turris/openwrt/-/issues/42

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>
---
 package/kernel/linux/modules/video.mk | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/package/kernel/linux/modules/video.mk b/package/kernel/linux/modules/video.mk
index 394dff6239..6cbec93419 100644
--- a/package/kernel/linux/modules/video.mk
+++ b/package/kernel/linux/modules/video.mk
@@ -855,6 +855,19 @@ endef
 
 $(eval $(call KernelPackage,video-gspca-sq905c))
 
+define KernelPackage/video-gspca-sq930x
+  TITLE:=sq930x webcam support
+  KCONFIG:=CONFIG_USB_GSPCA_SQ930X
+  FILES:=$(LINUX_DIR)/drivers/media/$(V4L2_USB_DIR)/gspca/gspca_sq930x.ko
+  AUTOLOAD:=$(call AutoProbe,gspca_sq930x)
+  $(call AddDepends/camera-gspca)
+endef
+
+define KernelPackage/video-gspca-sq930x/description
+ The SQ Technologies SQ930X based USB Camera Driver (sq930x) kernel module
+endef
+
+$(eval $(call KernelPackage,video-gspca-sq930x))
 
 define KernelPackage/video-gspca-stk014
   TITLE:=stk014 webcam support
-- 
2.30.2

